src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -710,7 +771,7 @@ public void Dispose()
         // internal.
         //
 
-        private static AsyncLocal<SafeAccessTokenHandle> s_currentImpersonatedToken = new AsyncLocal<SafeAccessTokenHandle>(CurrentImpersonatedTokenChanged);
+        internal static AsyncLocal<SafeAccessTokenHandle> s_currentImpersonatedToken = new AsyncLocal<SafeAccessTokenHandle>(CurrentImpersonatedTokenChanged);
```

Comment:
I don't see any reassignment of this field; can it be made `readonly`?  If not, I'd rather make an internal accessor for the tests than give them write access to a field.

Reply:
(tests, peer type, whatever; it still seems like the `readonly` modifier can be applied, since `AsyncLocal` is a reference type that never gets reassigned, only mutated)